uipa_org/settings/development.py

```diff
@@ -12,12 +14,15 @@ class Dev(UipaOrgThemeBase):
 
     CACHES = {"default": {"BACKEND": "django.core.cache.backends.dummy.DummyCache"}}
 
+    ELASTICSEARCH_HOST = values.Value("localhost", environ=True, environ_name="ELASTICSEARCH_HOST")
+    ELASTICSEARCH_USER = values.Value("elastic", environ=True, environ_name="ELASTICSEARCH_USER")
+    ELASTICSEARCH_PASSWORD = values.Value("****", environ=True, environ_name="ELASTICSEARCH_PASSWORD")
```

Comment:
Setting these ELASTICSEARCH_* variables in the shell environment before running `docker compose` commands works.

The `GDAL_LIBRARY_PATH` and `GEOS_LIBRARY_PATH` variables should be added here to make it more convenient for anyone that exec's into the app container to try things.

Reply:
I don't think we should build the workflow around using `docker compose`.  The container and code should allow these to be injected into the containers by whatever means and then work with that.  The uipa/django container may end up being used without the ES and postgres containers so we shouldn't assume that they're going to be around.

With the GDAL and GEOS env variables, I'm not sure what putting that in the `development.py` would get us.  Those settings get used on module import which happens quite a bit before `development.py` is evaluated.

Reply:
What I meant by:
> Setting these ELASTICSEARCH_* variables in the shell environment before running docker compose commands works.

was to confirm that "injecting" the settings works.

The GDAL and GEOS env variables are needed to further this aim because locating these libraries is important to getting a running Django server. The goal should be to make it easy for newcomers to become productive as quickly as possible.

I think that if we are providing containerization setup, we should describe the scenario that a user would go through in some documentation so that it can help users. This would also help reconcile the differences between running the Django server (`python manage.py runserver`) in a container and at the OS shell command prompt.

There's also the production deployment scenario that must be addressed so that switching between the various scenarios is easy as can be. Of course, we don't have any idea what the production deployment is actually like so we should be investigating that now if we are to have any success in this project.

Reply:
I've updated the code for this.  However, the location of these libraries won't change in the container so there shouldn't change.  We should document this for devs although I think most devs should probably use the containers for dev.  The production deployment should be using the container so the env will be the same.